frontend/purposeful-frontend/utils/fetch_wrapper.js

```diff
@@ -0,0 +1,130 @@
+// This file contains the fetch wrapper functions
+
+// Backend URL
+const BACKEND = "http://127.0.0.1:8080";
+
+/**
+ * Method to login a user given its username and password
+ * and store the token in local storage.
+ * @param {*} username - username of the user
+ * @param {*} password - password of the user, not encoded
+ * @returns true if login was successful, false otherwise
+ */
+export async function login(username, password) {
+  const encodedCreds = Buffer.from(`${username}:${password}`).toString(
+    "base64"
+  );
+
+  try {
+    const res = await fetch(`${BACKEND}/api/login`, {
+      method: "POST",
+      headers: { Authorization: "Basic " + encodedCreds },
+    });
+    const { status, ok } = await res;
+    if (!ok) {
+      console.error(
+        `Unable to get token. Status Code: ${status} Message: ${res.statusText}`
+      );
+      return false;
+    }
+    let token = await res.text();
+    // Store token in local storage
+    localStorage.setItem("token", token);
+    return true;
+  } catch (err) {
+    console.error("Something went wrong. Unable to get token. " + err);
+    return false;
+  }
+}
+/**
+ * Method to verify the validity of a token.
+ * @returns true if token is valid, false otherwise
+ */
+export async function verifyToken() {
+  const token = localStorage.getItem("token");
+  // Check if there is a token
+  if (!token) return false;
+  let appUserDTO = await fetchWrapper(`${BACKEND}/api/login`);
+  // Problem with the request
+  if (!appUserDTO) return false;
+  // Populate local storage with user info
+  localStorage.setItem("email", appUserDTO.email);
```

Comment:
Instead of using local storage, I believe it may be more appropriate to use session storage. What do you think.

Reply:
Local storage sounds good!